packages/colorpickers/src/elements/ColorPicker/ColorWell.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useRef, useState, useContext } from 'react';
+import { ThemeContext } from 'styled-components';
+import throttle from 'lodash.throttle';
+import { IHSVColor } from './reducer';
+import { hsl2hsv } from '../../utils/conversion';
+import { calculateNextHsv, getSaturationPosition } from '../../utils/saturation';
+import {
+  StyledColorWell,
+  StyledColorWellGradient,
+  StyledColorWellThumb,
+  StyledColorWellWrapper
+} from '../../styled';
+
+interface IColorWellProps {
+  hue: number;
+  saturation: number;
+  lightness: number;
+  onChange?: (hsv: IHSVColor, event: React.MouseEvent<any>) => void;
+}
+
+export const ColorWell: React.FC<IColorWellProps> = ({ hue, saturation, lightness, onChange }) => {
+  const { rtl } = useContext(ThemeContext);
+  const container = useRef<HTMLDivElement>(null);
+  const hsv = hsl2hsv(hue, saturation, lightness);
+  const [position, setPosition] = useState<{ s: number; v: number }>({
+    s: rtl ? 100 - hsv.s : hsv.s,
+    v: hsv.v
+  });
+
+  const throttledChange = throttle(e => {
+    if (container.current) {
+      const nextHsv = calculateNextHsv(e, hsv, container.current, rtl);
+      const saturationPosition = getSaturationPosition(e, container.current);
+
+      setPosition(saturationPosition);
+      onChange && onChange(nextHsv, e);
+    }
+  }, 50);
```

Comment:
I wasn't able to re-create the unexpected calls, could you outline the steps to re-create the unexpected calls so I could investigate further?

Reply:
With the throttle being at 50ms this might not be something that is easy to reproduce. I would still lean toward using `useRef` or `useCallback` to memorize this as mentioned in the stackoverflow thread.